vortex-array/src/array/struct/mod.rs

```diff
@@ -98,6 +99,41 @@ impl StructArray {
     }
 }
 
+impl StructArray {
+    /// Return a new StructArray with the given projection applied.
+    ///
+    /// Projection does not copy data arrays. Projection is defined by an ordinal array slice
+    /// which specifies the new ordering of columns in the struct. The projection can be used to
+    /// perform column re-ordering, deletion, or duplication at a logical level, without any data
+    /// copying.
+    ///
+    /// This function will return an error if the projection includes invalid column IDs.
+    pub fn project(self, projection: &[usize]) -> VortexResult<Self> {
+        let mut children = Vec::with_capacity(projection.len());
+        let mut names = Vec::with_capacity(projection.len());
+
+        let validity = self.validity().take(
```

Comment:
I don't think this is right? Struct validity is actually quite rare, not sure we ever use it, but it should be preserved untouched in this operation.

Reply:
yep, realized this right after i pushed. fixed now.